examples/micronaut-client/build.gradle

```diff
@@ -1,75 +1,40 @@
-    id "net.ltgt.apt-idea" version "0.21"
+    id("groovy")
+    id("io.micronaut.application") version "1.2.0"
+    id("com.google.cloud.tools.jib") version "2.6.0"
 }
 
-configurations {
-    // for dependencies that are needed for development only
-    developmentOnly
+repositories {
+    mavenCentral()
+    jcenter()
 }
 
-dependencies {
-    annotationProcessor platform("io.micronaut:micronaut-bom:$micronautVersion")
-    annotationProcessor "io.micronaut:micronaut-inject-java"
-    annotationProcessor "io.micronaut:micronaut-validation"
-    implementation platform("io.micronaut:micronaut-bom:$micronautVersion")
-    implementation "io.micronaut:micronaut-http-client"
-    implementation "io.micronaut:micronaut-http-server-netty"
-    implementation "io.micronaut:micronaut-inject"
-    implementation "io.micronaut:micronaut-validation"
-    implementation "io.micronaut:micronaut-runtime"
-    implementation 'io.micronaut:micronaut-management'
-
-    implementation project(':kubernetes-discovery-client')
-
-    runtime "ch.qos.logback:logback-classic:1.2.3"
-    testAnnotationProcessor platform("io.micronaut:micronaut-bom:$micronautVersion")
-    testAnnotationProcessor "io.micronaut:micronaut-inject-java"
-    testImplementation("org.spockframework:spock-core") {
-        exclude group: "org.codehaus.groovy", module: "groovy-all"
+micronaut {
+    runtime("netty")
+    testRuntime("spock2")
+    processing {
+        incremental(true)
+        annotations("micronaut.client.*")
     }
-    testImplementation "io.micronaut:micronaut-inject-groovy"
-    testImplementation platform("io.micronaut:micronaut-bom:$micronautVersion")
-    testImplementation "io.micronaut.test:micronaut-test-spock:$micronautTestVersion"
+}
 
+dependencies {
+    implementation("io.micronaut:micronaut-validation")
+    implementation("io.micronaut:micronaut-runtime")
+    implementation("javax.annotation:javax.annotation-api")
+    implementation("io.micronaut:micronaut-http-client")
+    implementation("io.micronaut:micronaut-management")
+    implementation project(':kubernetes-discovery-client')
+    runtimeOnly("ch.qos.logback:logback-classic")
     testCompile project(":test-utils")
 }
 
-shadowJar {
-    mergeServiceFiles()
+application {
+    mainClass.set("micronaut.client.Application")
 }
 
-test.classpath += configurations.developmentOnly
-
-mainClassName = "micronaut.client.Application"
-
-OperatingSystem os = DefaultNativePlatform.currentOperatingSystem
-
-jib {
-    from {
-//        image = 'gcr.io/distroless/java:debug'
-        image = 'alvarosanchez/alpine-with-curl'
-    }
-    to {
-        image = 'micronaut-kubernetes-example-client'
-        if (os.isMacOsX()) {
-            credHelper = 'osxkeychain'
-        }
+tasks {
+    dockerBuild {
+        images = ['micronaut-kubernetes-example-client']
     }
 }
-tasks.withType(JavaCompile){
-    options.encoding = "UTF-8"
-    options.compilerArgs.add('-parameters')
-}
-
-tasks.withType(JavaExec) {
```

Comment:
I guess these two are not needed anymore, right?

Reply:
Correct